Add generalized pip installation method (#1098)
fishtown-analytics/docs.getdbt.com: website/docs/docs/available-adapters.md

dbt connects to and runs SQL against your database, warehouse, platform, or query engine. It works by using a dedicated **adapter** for each technology. All the adapters listed below are open source and free to use, just like dbt.

### Installation

Most adapters can be installed from PyPi using `pip`. The installation will include `dbt-core` and any other required dependencies, which may include other adapter plugins. Read more about [installing dbt](dbt-cli/install/overview).

> Some vendor or community adapters may not exist in PyPi. However, an adapter hosted on GitHub can still be installed with `pip`:

Reviewer: @VersusFacit what do you think about defining the user input? Feel free to enhance my best guesses here:
```suggestion
Some vendor or community adapters may not exist in PyPi. However, you can still install an adapter hosted on GitHub with `pip install`, by replacing MAINTAINER_NAME with the person or company maintaining the adapter on GitHub and ADAPTER_NAME with the git repository's name (these can be taken directly from the package's url):
```

Author: I erred on the side of "let the code say it" as a guess of what to do. More than happy to go with an explicit definition of terms.

```shell

> pip install git+https://github.com/{MAINTAINER_NAME}/{ADAPTER_NAME}.git

Reviewer: I think we can just use the ALL CAPS and drop the curly brackets. 
```suggestion
pip install git+https://github.com/MAINTAINER_NAME/ADAPTER_NAME.git
```

| Athena                 | [Profile Setup](athena-profile)       | Athena engine version 2   | `pip install git+https://github.com/Tomme/dbt-athena.git` |

> Community-supported plugins are works in progress, and all users are encouraged to contribute by testing and writing code. If you're interested in contributing:

Reviewer 2: ```suggestion
Community-supported plugins are works in progress, and anyone is welcome to contribute by testing and writing code. If you're interested in contributing:
```
